envoy_repository_rule.bzl

```diff
@@ -0,0 +1,42 @@
+# Copyright 2019 Istio Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+def _envoy_repository_rule_impl(ctx):
+    """Core implementation of envoy_repository_rule."""
+
+    # Set repo, sha, sha256, and prefix from environment (if defined) or supplied attribute.
+    _repo = ctx.os.environ.get("ENVOY_REPOSITORY", default = ctx.attr.repository)
```

Comment:
Don't you need to add those variables to `.bazelrc` as well? I believe that we restrict values that we pass from the environment.

Reply:
The variables can be provided either in the `.bazelrc`, at `bazel` command execution or presumably using `BAZEL_BUILD_ARGS`. I do not have a concise set of pros/cons for each approach ATM.

Reply:
I guess my point was that if we add them to `.bazelrc` then those variables will be automatically picked from the environment.

Otherwise, yes, we can explicitly pass them to Bazel using `--action_env=`.

Reply:
I like the idea of utilizing `.bazelrc`. It appears to be the most straightforward of the various approaches. Let me add the update.